src/fuzzing/mutator.cc

```diff
@@ -0,0 +1,77 @@
+#include "src/fuzzing/mutator.h"
+
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include <random>
+
+#include "glog/logging.h"
+
+namespace fido2_tests {
+namespace {
+
+constexpr int kMutationOperations = 3;
+
+uint8_t RandomByte() { return static_cast<uint8_t>(std::rand() % 256); }
+
+}  // namespace
+
+Mutator::Mutator(int max_mutation_degree, int seed)
+    : max_mutation_degree_(max_mutation_degree) {
+  srand(seed);
+}
+
+bool Mutator::EraseByte(std::vector<uint8_t> &data, size_t max_size) {
+  if (data.size() <= 1) return false;
+  int index = std::rand() % data.size();
+  data.erase(data.begin() + index);
+  return true;
+}
+
+bool Mutator::InsertByte(std::vector<uint8_t> &data, size_t max_size) {
+  if (data.size() >= max_size) return false;
+  int index = std::rand() % data.size();
+  uint8_t elem = RandomByte();
+  data.insert(data.begin() + index, elem);
+  return true;
+}
+
+bool Mutator::ShuffleBytes(std::vector<uint8_t> &data, size_t max_size) {
```

Comment:
I'm currently reimplementing some of the LibFuzzers mutation operations. In our case, I think not shuffling the entire sequence makes sense since it's structured in different parts, and this can allow only shuffling data inside a structure without altering it.

Reply:
ah gotcha - so the start and end would be like shuffling one field value in the structure?

Reply:
Yes I have the hope that might happen sometimes